sktime/transformations/series/detrend/_detrend.py

```diff
@@ -28,26 +28,17 @@ class Detrender(_SeriesToSeriesTransformer):
     Depending on the passed data, this will require it to generate in-sample
     or out-of-sample forecasts.
 
-    The detrender also works in a pipeline as a form of boosting,
-    by first detrending a time series and then fitting another forecaster on
-    the residuals.
-
-    For example, to remove the linear trend of a time series:
-    forecaster = PolynomialTrendForecaster(degree=1)
-    transformer = Detrender(forecaster=forecaster)
-    yt = transformer.fit_transform(y_train)
-
     Parameters
     ----------
-    forecaster : estimator object, default=None
+    forecaster : sktime-like forecaster, default=None
```

Comment:
I would simply say `forecaster : sktime forecaster, follows BaseForecaster template, default = linear trend, i.e., PolynomialTrendForecaster(degree=1)`

Reply:
@aiwalter, minor comment on docstring - would you mind changing, then we can get this merged?